galaxy_importer/utils/execution_environment.py

```diff
@@ -0,0 +1,105 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import os
+import re
+
+from galaxy_importer.utils import yaml as yaml_utils
+
+
+def process_execution_environment(path, logger):
+    ex_env = {}
+    ex_env_path = os.path.join(path, 'meta', 'execution-environment.yml')
+
+    if os.path.exists(ex_env_path):
+        ex_env = _load_yaml(ex_env_path, logger)
+    else:
+        logger.info('No execution environment data found.')
+        return ex_env
+    if 'dependencies' not in ex_env:
+        logger.warning('No dependencies found')
+        return ex_env
+
+    python_exists = os.path.exists(os.path.join(path, ex_env['dependencies']['python']))
+    system_exists = os.path.exists(os.path.join(path, ex_env['dependencies']['system']))
+
+    if 'python' in ex_env['dependencies'] and python_exists:
```

Comment:
we are not handling the case where `python` is not in `ex_env['dependencies']` ... because of `ex_env['dependencies']['python']` this causes an exception:
```
version: 1
dependencies:
  system: bindep.txt
```